src/Scaler.Demo/OrderGenerator/Program.cs

```diff
@@ -0,0 +1,147 @@
+        }
+
+        private static int ReadOrderCount()
+        {
+            while (true)
+            {
+                Console.Write("Let's queue some orders, how many do you want? ");
+
+                if (int.TryParse(Console.ReadLine(), out int count) && count >= 1 && count <= 10000)
+                {
+                    return count;
+                }
+
+                Console.WriteLine("That's not a valid amount. Please enter a number between 1 and 10000.");
+            }
+        }
+
+        private static bool ReadIsSingleArticle()
+        {
+            Console.Write("Do you want to limit orders to single article (to put them in a single partition)? (Y/N) ");
+            bool isSingleArticle = Console.ReadKey().Key == ConsoleKey.Y;
+            Console.WriteLine();
+            return isSingleArticle;
+        }
+
+        private static async Task CreateOrdersAsync(int count, bool isSingleArticle)
+        {
+            Container container = new CosmosClient(_cosmosDbConfig.Connection)
+                .GetContainer(_cosmosDbConfig.DatabaseId, _cosmosDbConfig.ContainerId);
+
+            int remainingCount = count;
+            string article = isSingleArticle ? new Commerce().Product() : null;
+
+            while (remainingCount > 0)
+            {
+                // Do not push all orders together as that may cause requests to get throttled.
+                int newCount = Math.Min(remainingCount, 20);
+
+                Task[] createOrderTasks = Enumerable.Range(0, newCount)
+                    .Select(_ => CreateOrderAsync(container, article))
+                    .ToArray();
+
+                await Task.WhenAll(createOrderTasks);
+                await Task.Delay(TimeSpan.FromSeconds(2));
+
+                remainingCount -= newCount;
+            }
+
+            Console.WriteLine("That's it, see you later!");
+        }
+
+        private static async Task CreateOrderAsync(Container container, string article)
+        {
+            Customer customer = new Faker<Customer>()
+                .RuleFor(customer => customer.FirstName, faker => faker.Name.FirstName())
+                .RuleFor(customer => customer.LastName, faker => faker.Name.LastName());
+
+            Order order = new Faker<Order>()
+                .RuleFor(order => order.Customer, () => customer)
+                .RuleFor(order => order.Amount, faker => faker.Random.Number(1, 10))
+                .RuleFor(order => order.Article, faker => article ?? faker.Commerce.Product());
+
+            Console.WriteLine($"Creating order {order.Id} - {order.Amount} unit(s) of {order.Article} for {order.Customer.FirstName} {order.Customer.LastName}");
+            await container.CreateItemAsync(order);
+        }
+
+        private static async Task SetupAsync()
+        {
+            Console.WriteLine($"Creating database: {_cosmosDbConfig.DatabaseId}");
+
+            Database database = await new CosmosClient(_cosmosDbConfig.Connection)
```

Comment:
nit: Worth using a single CosmosClient instance instead of creating one in the Setup, another in CreateOrdersAsync and another in Teardown

Reply:
This isn't a problem, since in one run of the application, either `SetupAsync` or `CreateOrdersAsync` or `TeardownAsync` will execute (that too once only).